libtsuba/src/ParquetReader.cpp

```diff
@@ -71,70 +72,41 @@ HandleBadParquetTypes(std::shared_ptr<arrow::ChunkedArray> old_array) {
 }
 
 Result<std::unique_ptr<parquet::arrow::FileReader>>
-MakeFileReader(
-    const katana::Uri& uri, uint64_t preload_start, uint64_t preload_end,
-    std::shared_ptr<tsuba::FileView>* fv_ptr = nullptr) {
-  auto fv = std::make_shared<tsuba::FileView>(tsuba::FileView());
-  if (auto res = fv->Bind(uri.string(), preload_start, preload_end, false);
-      !res) {
-    return res.error().WithContext("opening {}", uri);
-  }
+BuildReader(
+    const std::string& uri, bool preload,
```

Comment:
Why use string `uri` here instead of `katana::Uri`?

Reply:
Oh it's a private function, and a bunch of places I call it with strings from `fmt::format` so it seemed silly to convert back and forth from known well formed uris.